pkg/performanceprofile/profilecreator/profilecreator.go

```diff
@@ -334,31 +334,53 @@ func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, sp
 }
 
 type cpuAccumulator struct {
-	builder *cpuset.Builder
-	count   int
+	elems map[int]struct{}
+	count int
+	done  bool
 }
 
 func newCPUAccumulator() *cpuAccumulator {
 	return &cpuAccumulator{
-		builder: cpuset.NewBuilder(),
+		elems: map[int]struct{}{},
+		count: 0,
+		done:  false,
 	}
 }
 
 // AddCores adds logical cores from the slice of *cpu.ProcessorCore to a CPUset till the cpuset size is equal to the max value specified
 // In case the max is specified as allCores, all the cores from the slice of *cpu.ProcessorCore are added to the CPUSet
-func (ca *cpuAccumulator) AddCores(max int, cores []*cpu.ProcessorCore) {
+func (ca *cpuAccumulator) AddCores(max int, cores []*cpu.ProcessorCore) int {
+	if ca.done {
+		return -1
+	}
+	initialCount := ca.count
 	for _, processorCore := range cores {
-		for _, core := range processorCore.LogicalProcessors {
+		for _, logicalProcessorId := range processorCore.LogicalProcessors {
 			if ca.count < max || max == allCores {
-				ca.builder.Add(core)
-				ca.count++
+				_, found := ca.elems[logicalProcessorId]
+				ca.elems[logicalProcessorId] = struct{}{}
+				if !found {
+					ca.count++
+				}
 			}
 		}
 	}
+	return ca.count - initialCount
+}
+
+func (ca *cpuAccumulator) mapKeys() []int {
+	keys := make([]int, len(ca.elems))
+	i := 0
+	for k := range ca.elems {
+		keys[i] = k
```

Comment:
maybe just use append? also, we can probably inline this function in `Result`
```
func (ca *cpuAccumulator) Result() cpuset.CPUSet {
        var keys []int
        for k := range ca.elems {
             keys = append(keys, k)
        }
	return cpuset.NewCPUSet(keys...)
}
```

Reply:
done